.dotnet/api/OpenAI.netstandard2.0.cs

```diff
@@ -280,7 +280,7 @@ public class AssistantCreationOptions : IJsonModel<AssistantCreationOptions>, IP
         BinaryData IPersistableModel<AssistantCreationOptions>.Write(ModelReaderWriterOptions options);
     }
     public class AssistantModificationOptions : IJsonModel<AssistantModificationOptions>, IPersistableModel<AssistantModificationOptions> {
-        public IList<ToolDefinition> DefaultTools { get; set; }
+        public IList<ToolDefinition> DefaultTools { get; }
```

Comment:
Thank you, Shivangi! After this change, the only collection properties left that have a setter are the following:

* MessageCreationOptions.Attachments
* AssistantCreationOptions.Metadata
* AssistantModificationOptions.Metadata
* MessageCreationOptions.Metadata
* MessageModificationOptions.Metadata
* RunModificationOptions.Metadata
* ThreadCreationOptions.Metadata
* ThreadModificationOptions.Metadata
* VectoreStoreCreationOptions.Metadata
* VectorStoreModificationOptions.Metadata

Paging in @trrwilson because we talked about something related a while ago. The properties above are getting a setter from the generator because they are explicitly specified as nullable in the TypeSpec (which itself comes from the swagger spec). Now, in basically every part of the REST API that I have encountered so far, setting a nullable property to `null` does not imply anything special and it's equivalent to not setting it at all. Assuming `null` doesn't mean anything special here either, I wonder if we should add a simple change to customize these properties to remove the setter in favor of consistency and a simpler API.

Reply:
@ShivangiReja : I have approved this PR, and you can merge it. If we decide to revise these other properties, they can go in a separate PR. 😊